Source/Core/DolphinQt/Debugger/CodeWidget.cpp

```diff
@@ -29,6 +29,8 @@ CodeWidget::CodeWidget(QWidget* parent) : QDockWidget(parent)
 {
   setWindowTitle(tr("Code"));
   setObjectName(QStringLiteral("code"));
+  
+  setHidden(!Settings::Instance().IsCodeVisible() || !Settings::Instance().IsDebugModeEnabled());
```

Comment:
`setVisible(Settings::Instance().IsCodeVisible() && Settings::Instance().IsDebugModeEnabled());` would be more readable I think. (I know you just moved the existing code.)

Change seems fine otherwise. It's strange that Qt is showing the window so early, I'm guessing `setAllowedAreas` is the problem function triggering the window to be created at that point?

Reply:
Should be noted that setHidden is used in 2 other places in Line 43 and 54 so do you want you keep it as-is, change all or just change this one line?

It is actually `setFloating(settings.value(QStringLiteral("codewidget/floating")).toBool());`. This bug is not an issue on Windows from my testing but it probably has to do with how macOS presents application windows.

Reply:
You can change them all to setVisible or leave them. Doesn't matter. Just figured it might as well be cleaned up.

Reply:
I will leave them for consistency as other files with similar functionality use `setHidden` in this way.